app/components/primer/alpha/action_list/item.rb

```diff
@@ -36,20 +36,31 @@ class Item < Primer::Component
         #
         # To render an icon, call the `with_leading_visual_icon` method, which accepts the arguments accepted by <%= link_to_component(Primer::Beta::Octicon) %>.
         #
-        # To render an avatar, call the `with_leading_visual_avatar` method, which accepts the arguments accepted by <%= link_to_component(Primer::Beta::Avatar) %>.
-        #
         # To render an SVG, call the `with_leading_visual_svg` method.
         #
         # To render custom content, call the `with_leading_visual_content` method and pass a block that returns a string.
         renders_one :leading_visual, types: {
-          icon: Primer::Beta::Octicon,
-          avatar: ->(**kwargs) { Primer::Beta::Avatar.new(**{ **kwargs, size: 16 }) },
+          icon: lambda { |**system_arguments, &block|
+            deny_aria_key(
+              :label,
+              "Avoid using `aria-label` on leading visual icons, as they are purely decorative.",
+              **system_arguments
+            )
+
+            Primer::Beta::Octicon.new(**system_arguments, &block)
+          },
+          avatar: lambda { |*|
+            return unless should_raise_error?
+
+            raise "Leading visual avatars are no longer supported. Please use the #with_avatar_item slot instead."
+          },
           svg: lambda { |**system_arguments|
             Primer::BaseComponent.new(tag: :svg, width: "16", height: "16", **system_arguments)
           },
           content: lambda { |**system_arguments|
             Primer::BaseComponent.new(tag: :span, **system_arguments)
-          }
+          },
+          raw_content: nil
```

Comment:
What is this for?

Reply:
Good question. The existing `content` slot wraps the content in a `<div>` via `BaseComponent`. I tried to use it at first but it caused a subtle padding issue with avatars and failed the visual regressions CI job. I added this new `raw_content` slot to be able to pass in arbitrary content. The `nil` tells ViewComponent to render the content directly instead of a lambda or component instance.